src/main/java/org/betonquest/betonquest/objectives/EnchantObjective.java

```diff
@@ -46,15 +49,27 @@ public void onEnchant(final EnchantItemEvent event) {
         if (!item.compare(event.getItem())) {
             return;
         }
-        for (final EnchantmentData enchant : enchantments) {
-            if (!event.getEnchantsToAdd().containsKey(enchant.getEnchantment())
-                    || event.getEnchantsToAdd().get(enchant.getEnchantment()) < enchant.getLevel()) {
-                return;
-            }
+
+        if (matchesDesiredEnchants(event.getEnchantsToAdd()) && checkConditions(onlineProfile)) {
+            getCountingData(onlineProfile).progress();
+            completeIfDoneOrNotify(onlineProfile);
         }
-        if (checkConditions(onlineProfile)) {
-            completeObjective(onlineProfile);
+    }
+
+    private boolean matchesDesiredEnchants(final Map<Enchantment, Integer> addedEnchants) {
+        boolean hasEnchant = false;
+
+        for (final EnchantmentData enchant : desiredEnchantments) {
+            final Enchantment desiredEnchant = enchant.enchantment();
+            final int desiredLevel = enchant.level();
+            if (addedEnchants.containsKey(desiredEnchant)) {
+                final int addedLevel = addedEnchants.get(desiredEnchant);
+                if (addedLevel >= desiredLevel) {
+                    hasEnchant = true;
+                }
+            }
         }
+        return hasEnchant;
```

Comment:
This is an `or` and not an `and` implementation. I think it should be and.

Reply:
So we should check if the desired enchantment is already on the item used in the event?

Reply:
The default is now an `and` implementation as it seems like that was the intended previous behaviour.
This behavior can be changed by setting the optional `requirementMode` argument to `one`.